API/Controllers/UserController.cs

```diff
@@ -11,81 +11,63 @@
 using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Mvc;
-
+using Services.Interface;
 
 namespace API.Controllers
 {
     [Authorize]
     public class UsersController : BaseApiController
     {
-        private readonly IUserRepository _userRepository;
-        private readonly IMapper _mapper;
+        private readonly IUserService _userService;
         private readonly IPhotoService _photoService;
+        private readonly IMapper _mapper;
         public UsersController(
-            IUserRepository userRepository,
-            IMapper mapper,
-            IPhotoService photoService
+            IUserService userService,
+            IPhotoService photoService,
+            IMapper mapper
         )
         {
-            _userRepository = userRepository;
-            _mapper = mapper;
+            _userService = userService;
             _photoService = photoService;
+            _mapper = mapper;
         }
 
         [HttpGet]
         public async Task<ActionResult<IEnumerable<MemberDTO>>> GetUsers()
         {
-            var users = await _userRepository.GetMembersAsync();
+            var users = await _userService.GetUsers();
 
             return Ok(users);
         }
 
         [HttpGet("{username}")]
         public async Task<ActionResult<MemberDTO>> GetUser(string username)
         {
-           return await _userRepository.GetMemberAsync(username);
+           return await _userService.GetUser(username);
         }
 
         [HttpPut]
         public async Task<ActionResult> UpdateUser(MemberUpdateDTO memberUpdateDTO)
         {
-            var user = await _userRepository.GetUserByUsernameAsync(User.GetUsername());
+            bool isUpdated = await _userService.UpdateUser(memberUpdateDTO, User.GetUsername());
 
-            _mapper.Map(memberUpdateDTO, user);
-
-            _userRepository.Update(user);
-
-            if (await _userRepository.SaveAllAsync()) { return NoContent(); }
+            if (isUpdated) { return NoContent(); }
 
             return BadRequest("Failed to update user");
         }
 
         [HttpPost("add-photo")]
         public async Task<ActionResult<PhotoDTO>> AddPhoto(IFormFile file)
         {
-            var user = await _userRepository.GetUserByUsernameAsync(User.GetUsername());
+            var user = await _userService.GetUserByUsername(User.GetUsername());
```

Comment:
Good to see you wrap the userRepository into the userService, and you can also inject the PhotoService into the UserService. As generally, we don't have the business logic in the controller